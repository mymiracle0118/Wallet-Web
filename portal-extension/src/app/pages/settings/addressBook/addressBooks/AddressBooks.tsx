import React from 'react'
import { useTranslation } from 'react-i18next'
import { useSettings } from '@portal/shared/hooks/useSettings'

import SinglePageTitleLayout from 'layouts/single-page-layout/SinglePageLayout'
import { Button } from 'components'
import AddressItem from '../AddressItem'
import SupraIcon from 'assets/networks/Supra.svg'
import { useNavigate } from 'lib/woozie'

const AddressBooks = () => {
  const { navigate } = useNavigate()
  const { t } = useTranslation()
  const { addressBook } = useSettings()

  return (
    <SinglePageTitleLayout dataAid="addressBookHead" title={t('Settings.addressBookPageTitle')}>
      <Button
        data-aid="editButton"
        variant="bordered"
        color="outlined"
        className="mb-4"
        onClick={() => navigate(`/settings/edit-address`)}
      >
        {t('Actions.edit')}
      </Button>
      {addressBook.map(
        (data, idx) =>
          data.username &&
          data.address && (
            <AddressItem key={idx} image={<SupraIcon />} userName={`@${data.username}`} address={data.address} />
          )
      )}
    </SinglePageTitleLayout>
  )
}

export default AddressBooks
