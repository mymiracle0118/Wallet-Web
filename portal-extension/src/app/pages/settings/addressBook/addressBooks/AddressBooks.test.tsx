import React from 'react'
import renderer from 'react-test-renderer'

import AddressBooks from './AddressBooks'

describe('<AddressBooks />', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<AddressBooks />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
