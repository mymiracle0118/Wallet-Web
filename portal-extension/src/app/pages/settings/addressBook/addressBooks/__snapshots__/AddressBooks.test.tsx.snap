// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressBooks /> renders correctly 1`] = `
<div
  className="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
>
  <div
    className="flex flex-row relative justify-between items-center h-16"
  >
    <button
      className="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
      onClick={[Function]}
      type="button"
    >
      <arrow-left.svg
        className="text-xl stroke-custom-black dark:stroke-custom-white"
      />
    </button>
    <div
      className="flex-1 text-center"
    >
      <div
        className="text-custom-black dark:text-custom-white text-lg font-extrabold"
        data-aid="addressBookHead"
        style={
          Object {
            "WebkitLineClamp": "3",
          }
        }
      >
        Settings.addressBookPageTitle
      </div>
    </div>
  </div>
  <div
    className="w-full h-full overflow-auto"
  >
    <div
      className="px-4"
    >
      <button
        className="mb-4 flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full h-[44px] border-[1px] border-solid text-primary dark:text-custom-white border-primary dark:border-custom-white40 dark:hover:border-custom-white hover:border-primary-alt hover:text-primary-alt"
        data-aid="editButton"
        onClick={[Function]}
        type="button"
      >
        Actions.edit
      </button>
    </div>
  </div>
</div>
`;
